src/CowSdk.ts

```diff
@@ -1,31 +1,42 @@
 import { Signer } from 'ethers'
-import log, { LogLevelDesc } from 'loglevel'
+import log from 'loglevel'
 import { CowError } from './utils/common'
 import { CowApi, CowSubgraphApi, MetadataApi } from './api'
-import { SupportedChainId as ChainId, SupportedChainId } from './constants/chains'
+import { SupportedChainId as ChainId } from './constants/chains'
 import { Context, CowContext } from './utils/context'
 import { signOrder, signOrderCancellation, UnsignedOrder } from './utils/sign'
 import { ZeroXApi } from './api/0x'
-import { ZeroXOptions } from './api/0x/types'
 import ParaswapApi from './api/paraswap'
-import { ParaswapOptions } from './api/paraswap/types'
-import { WithEnabled } from './types'
+// types
+import {
+  Options,
+  OptionsWithApisEnabledStatus,
+  OptionsWithParaswapEnabled,
+  OptionsWithZeroXEnabled,
+  ParaswapEnabled,
+  ZeroXEnabled,
+} from 'sdk'
+import { ExtendsObject } from 'utilities'
 
-type Options = {
-  loglevel?: LogLevelDesc
-  zeroXOptions?: Partial<ZeroXOptions & WithEnabled>
-  paraswapOptions?: Partial<ParaswapOptions> & WithEnabled
-}
-
-export class CowSdk<T extends ChainId> {
+interface ICowSdk<Opt extends Options> {
```

Comment:
are we going to keep the `I` prefix for interfaces? I'm fine with it as I came from c#, but we should define the guideline and keep it throughout the repos. Also, about the interfaces vs types debate, do you have a strong opinion about it? Like interface for classes and types for the rest?

Reply:
so I decided against the Interface approach to keep the class simpler - it's just a personal preference for me, the interface I wrote as a test and was lazy in changing it back.

however if you can argue a solid reason to change these across the board I am not against it

Reply:
Well, to be fair, this is the only place I ever saw the `I` prefix for interfaces in our code base.
I second Ramiro and would prefer not to have it

Reply:
Regarding the debate of types for this and interfaces for that, it's a mess everywhere so that should stay as is to avoid a 5k files change in all our repos =P

Reply:
(But I do favor types everywhere except where interfaces are required and do that for my own code)